crates/iota-indexer/src/historical_fallback/convert.rs

```diff
@@ -0,0 +1,227 @@
+// Copyright (c) 2025 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! Provides helper functions and types for converting historical fallback
+//! types into Indexer-compatible types for the JSON-RPC API.
+
+use std::sync::Arc;
+
+use iota_json_rpc_types::{IotaEvent, IotaTransactionKind};
+use iota_package_resolver::{PackageStore, Resolver};
+use iota_types::{
+    digests::TransactionDigest,
+    dynamic_field::DynamicFieldType,
+    effects::{TransactionEffects, TransactionEffectsAPI, TransactionEvents},
+    messages_checkpoint::{
+        CertifiedCheckpointSummary, CheckpointContents, CheckpointSequenceNumber,
+    },
+    object::Object,
+    transaction::{Transaction, TransactionDataAPI},
+};
+use prometheus::Registry;
+
+use crate::{
+    errors::IndexerResult,
+    ingestion::{common::prepare::extract_df_kind, primary::prepare::InMemTxChanges},
+    metrics::IndexerMetrics,
+    models::{
+        checkpoints::StoredCheckpoint,
+        objects::StoredObject,
+        transactions::{StoredTransaction, tx_events_to_iota_tx_events},
+    },
+    types::{IndexedCheckpoint, IndexedObject, IndexedTransaction},
+};
+
+/// Represents the data needed to fetch from historical fallback storage in
+/// order to convert it into compatible types the indexer can use for JSON RPC
+/// API.
```

Comment:
Is it intended that the same comment is duplicated before every HistoricalFallback* type? This was a bit strange to read, maybe we could at least mention in the comment that this is for the `Object` type?

The same goes for all other types.